ai-development-guidelines-v2.md

```diff
@@ -0,0 +1,375 @@
+# AI-Assisted Development: Human-in-the-Loop Guidelines
+
+## Executive Summary
+
+This document establishes comprehensive guidelines for integrating AI code assistants (claude.code, GitHub Copilot, etc.) into software development workflows while maintaining critical human oversight and quality control.
+
+## 1. Development Workflow Requirements
+
+### 1.1 Pre-Development Phase
+
+**AI Usage Declaration**
+- Engineers must declare AI assistant usage in commit messages using standardized tags:
+  - `[AI-ASSISTED]` - AI helped with implementation
+  - `[AI-GENERATED]` - Majority of code generated by AI
+  - `[AI-REVIEWED]` - AI used for code review suggestions
+
+**Human Validation Checkpoints**
+- **Requirement Analysis**: Human must validate AI's understanding of requirements before implementation
+- **Architecture Review**: Senior engineer approval required for AI-suggested architectural decisions
+- **Security Impact Assessment**: Human security review mandatory for any AI-generated code touching:
+  - Authentication/authorization
+  - Data validation
+  - External APIs
+  - Database operations
+
+### 1.2 Development Phase Guardrails
+
+**Code Generation Rules**
+1. **Maximum AI Block Size**: No single AI-generated code block exceeds 50 lines without human review
+2. **Critical Path Restriction**: AI cannot generate code for:
+   - Payment processing
+   - User authentication
+   - Data encryption/decryption
+   - Database migration scripts
+   - Production deployment scripts
+
+**Human Oversight Requirements**
+- **Test Coverage**: Human-written tests required for all AI-generated functions
+- **Documentation**: Human must write/verify all documentation for AI-assisted code
+
+## 2. Pull Request (PR) Review Process
+
+### 2.1 Enhanced PR Template
+
+```markdown
+## AI Assistant Usage
+- [ ] AI assistant used (specify which: claude.code, Copilot, etc.)
+- [ ] Code blocks generated by AI are clearly marked
+- [ ] All AI-generated code has been human-reviewed and understood
+- [ ] Tests written/reviewed by human for AI-generated code
+
+## Human Validation Checklist
+- [ ] Requirements correctly interpreted
+- [ ] Architecture patterns followed
+- [ ] Security considerations addressed
+- [ ] Performance implications considered
+- [ ] Error handling implemented
+- [ ] Edge cases covered
+
+## AI-Generated Code Blocks
+List specific files/functions/classes that were AI-generated:
+- File: `src/utils/helper.js`, Lines: 15-45, AI: claude.code
+- Function: `validateUserInput()`, AI: GitHub Copilot
+```
+
+### 2.2 Review Assignment Rules
+
+**Mandatory Human Reviewers**
+- **All PRs**: Minimum 2 human reviewers (1 senior engineer)
```

Comment:
This is hopefully already the case - or at least the PRs are reviewed by at least one subject matter expert.